### PR TITLE
Add documentation for personal access tokens

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Add documentation for Personal Access Tokens

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/14910

## Release Notes

```release-note
Add documentation for personal access tokens

```

/hold

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/3081"></a>

### CODE REVIEW

**Comment by Siddhant-K-code** [comment]
WDYT, we should release this public API (#2939)?

**Comment by easyCZ** [comment]
The PATs are independent, and a pre-requisite for the Pub API. I think we should release this first, and then follow-up with the Pub API docs

File: gitpod/docs/configure/user-settings/index.md [hunk 1]
```diff
@@ -16,3 +16,4 @@ User settings allow you to customize your own personal Gitpod experience.
 - [Browser Bookmarklet](/docs/configure/user-settings/browser-bookmarklet)
 - [Dotfiles](/docs/configure/user-settings/dotfiles)
 - [SSH](/docs/configure/user-settings/ssh)
+- [Access Tokens](/docs/configure/user-settings/access-tokens)
```
Comment (gtsiolis) [inline gitpod/docs/configure/user-settings/index.md:19]: **issue**: It makes sense to also add this in the left navigation sidebar, right?

```diff
diff --git a/src/lib/contents/docs/menu.ts b/src/lib/contents/docs/menu.ts
index 3ff5d52a..8ec729ab 100644
--- a/src/lib/contents/docs/menu.ts
+++ b/src/lib/contents/docs/menu.ts
@@ -64,6 +64,7 @@ export const MENU: MenuEntry[] = [
     M("Browser bookmarklet", "configure/user-settings/browser-bookmarklet"),
     M("Dotfiles", "configure/user-settings/dotfiles", false, []),
     M("SSH", "configure/user-settings/ssh"),
+    M("Access Tokens", "configure/user-settings/access-tokens"),
   ]),
 
   M(
```

File: gitpod/docs/configure/user-settings/access-tokens.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+By default, an Access Token has **No Access**. You need to explicitly select Full Access to succesfully authorize.
+
+In the future, we will expand Access Tokens with fine grained permissions.
+
+## Expiration
+
+All Access Tokens have an expiration time. When a token expires, requests made with the token will fail with Unathorized. When a token expires, you can [regenerate](#regenerating-access-tokens) it.
+
+## Managing Access Tokens
```
Comment (gtsiolis) [inline gitpod/docs/configure/user-settings/access-tokens.md:30]: **praise**: Thanks for :a: listing all _actions_ a user can take on tokens and also :b: adding separate sections for all the token _attributes_.

File: gitpod/docs/configure/user-settings/access-tokens.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+![List Access Tokens](../../../static/images/docs/access-token-list.png)
+
+### Creating an Access Token
+
+You can create an Access Token by navigating to [gitpod.io/tokens](https://gitpod.io/tokens) or through Settings > Access Tokens. Clicking **New Access Token** will prompt you to specify details of the token.
+
+- Your access tokens will automatically expire after the specified duration.
+- Once you create an Access Token, it will be shown to you **once**. Make sure you store it in a secure place, you will not be able to retrieve it again.
+
+![Create Access Token](../../../static/images/docs/access-token-create.png)
```
Comment (gtsiolis) [inline gitpod/docs/configure/user-settings/access-tokens.md:45]: **issue(non-blocking)**: Although screenshots in docs can be useful sometimes, I'd like to resurface some relevant feedback for future reference, as screenshots in this PR are already obsolete for user users behind the UBP feature flag, see https://github.com/gitpod-io/website/pull/2657#discussion_r959590384.


> This is another proof that using screenshots can help, however, they could be used sparingly as they a) tend to become out-of-date easily, b) are expensive to maintain, and c) are not as accessible as text.

> I’d think most of the existing screenshots in our docs 🅰️ are already obsolete [[1](https://www.gitpod.io/docs/gitlab-integration)][[2](https://www.gitpod.io/docs/ides-and-editors/intellij)][[3](https://www.gitpod.io/docs/ides-and-editors/vscode)][…] and 🅱️ can be easily replaced with text or bullet points that describe any necessary steps.

> Sometimes an image is worth 1000 words! I’m not suggesting removing all images but sparingly use them.

> From a technical writing perspective, we first needs some guidelines like capturing only the relevant parts, be consistent, etc. Currently, there’s a big range of differences in screenshots, themes used, image dimensions or resolution,  etc. Minor but it makes the docs feel like a second-class citizen.

> Side-by-side images should be avoided at all costs. It’s a good sign of either poor UX for a product area or low documentation quality.

> Last but not least, including this here for reference, I still find the [Teams & Projects](https://www.gitpod.io/docs/teams-and-projects) docs section a good example of documentation containing imperative guidelines of how to do things, future-proof content even while the feature is evolving, probably some of the most anchor-linked sections that can be linked also from within the dashboard itself[[1](https://www.gitpod.io/docs/team-billing#:~:text=New%20members%20are%20added%20using%20a%20shared%20invite%20URL%20which%20can%20be%20reset.)][[2](https://www.gitpod.io/docs/environment-variables#:~:text=Environment%20variables%20which%20are%20defined%20in%20project%20settings)][..].

| With UBP feature flag | Without UBP feature flag |
|-|-|
| nav-before | nav-after |

**Review by gtsiolis — COMMENTED** [review]
ok

File: gitpod/docs/configure/user-settings/index.md (hunk 1 above)
Comment (easyCZ) [reply gitpod/docs/configure/user-settings/index.md:19]: Ah, that's where it goes. Wasn't able to find it.

File: gitpod/docs/configure/user-settings/access-tokens.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+## Expiration
+
+All Access Tokens have an expiration time. When a token expires, requests made with the token will fail with Unathorized. When a token expires, you can [regenerate](#regenerating-access-tokens) it.
+
+## Managing Access Tokens
+
+### Listing Access Tokens
+
+You can list your Access Tokens by going to [gitpod.io/tokens](https://gitpod.io/tokens).
```
Comment (jldec) [inline gitpod/docs/configure/user-settings/access-tokens.md:34]: Suggest mentioning that this is "in your user settings".

Comment (gtsiolis) [reply gitpod/docs/configure/user-settings/access-tokens.md:34]: +1

Comment (easyCZ) [reply gitpod/docs/configure/user-settings/access-tokens.md]: Added.

**Review by jldec — APPROVED** [review]
This looks good.

My only small suggestion would be to move the 2nd screenshot (New Access Token) to  the top above the ## Permissions header, so that it can serve as a visual guide for those users who just like to look and learn.

**Comment by easyCZ** [comment]
@jldec That feels odd, as the First section describes what a Token is, and the properties and only subsequently we show the operations that can be done on the tokens. If you still think it makes sense, I'll move it. Let me know.

**Comment by easyCZ** [comment]
> My only small suggestion would be to move the 2nd screenshot (New Access Token) to the top above 

Moved the image.

**Pull request merged**